cmd/root.go

```diff
@@ -372,6 +389,14 @@ func initConfig() {
 	if len(dataDirectory) != 0 {
 		Config.DataDirectory = dataDirectory
 	}
+
+	if allInMemory {
+		Config.AllInMemoryEnabled = allInMemory
+	}
+
+	if tableSize != 0 {
```

Comment:
what if I provide a -ve value - would it break? do we need to  make sure we put a check on negative value?

Reply:
if the number not in our range, we will use the default value, for L0 in memory mode is 3, for all in meory is 6, the range is [2,100], so <= 0 here means using default number

Reply:
no I meant to see what if a user provide -ve value as input. Will our tool break?

Reply:
I @ you in line 195, for the change here, we change the config file value. 
and before loading config file's tableSize, we do a value checking
BTW, if users provide nagetive number, not break, and we will not tell the user, just using default value silencely

Reply:
okay.  just want to make sure it doesn't break anything.

Reply:
I don't think that's a good way from error handling perspective. In our validation, we should error out invalid user input directly instead of silently using any value that user is not aware of explicitly. This can cause user confuse especially the negative value is typeo.

Reply:
@GeekArthur currently all the command flag checking for CLI seems are silently, and modified the data silently. perhaps sometimes is for switch to default value. 
We could figure out there later:

-  it is there any methods to omit a command line flag instead of require a default value 
-  it is there any way to throw errors in configuration file

Reply:
@jingweicb 
> currently all the command flag checking for CLI seems are silently, and modified the data silently
Can you provide some cases in our existing CLI? IMO, the CLI tool without flag/argument validation shouldn't be released

The cobra package does support build-in flag/argument validation. You may refer to https://github.com/kubernetes/kubernetes

Also isn't the in-memory config in config file? Why does CLI flag related to it?

Reply:
@GeekArthur DM you soon !